infogami/infobase/_dbstore/save.py

```diff
@@ -332,7 +332,7 @@ def _dict_difference(self, d1, d2, eq=None):
         """Returns set equivalant of d1.difference(d2) for dictionaries.
         """
         eq = eq or (lambda a, b: a == b)
-        return dict((k, v) for k, v in d1.iteritems() if not eq(v, d2.get(k)))
+        return {k: v for k, v in d1.items() if not eq(v, d2.get(k))}
```

Comment:
Used only once.  Python 2 will build a list in memory which will take up more RAM.  Python 3 will be fast.